src/instruction.cpp

```diff
@@ -302,6 +302,24 @@ void instruction::replace_mod_argument(module_ref old, module_ref new_mod)
     std::replace(module_args.begin(), module_args.end(), old, new_mod);
 }
 
+bool instruction::is_undefined() const
+{
+    if(op.name() == "undefined")
+    {
+        return true;
+    }
+    else if(is_context_free(op))
```

Comment:
Remove the `is_context_free` check here. And just do an `else` and the check all of the inputs.